archivesunleashed/auk
# Implement spark.executor.heartbeatInterval option, resolves #128

**GitHub issue(s)**:

Resolves #128 

# What does this Pull Request do?

This pull request does the following:
* adds the `--conf spark.executor.heartbeatInterval` flag to the Spark job;
* adds a sample timeout of `600s` to the example application.yml file.

This will help make AUT more stable when working with the very occasional WARC record. We haven't encountered too many issues, but once in a while we run into something weird that takes a long time for our job to crunch through. 

`--conf spark.executor.heartbeatInterval` is by default set to `10s`. This increases it to `600s`. As a result, 739/4532 which was previously failing is now working.

# How should this be tested?

I have tested this on broken collections on `rho`, and have also done some basic smoke testing to make sure nothing else has gone awry. That being said, I haven't tested this on our production server. 

Nothing should break, but it would be worth testing on 739/4532 to make sure it's working on the canonical AUK instance.

# Interested parties

@ruebot – **note that this isn't critical and can wait until after vacations are all done.** 😄 

Thanks in advance for your help with the Archives Unleashed Toolkit!
